make_prg/from_msa/cluster_sequences.py

```diff
@@ -20,6 +18,33 @@
 ClusteredSeqs = List[Sequences]
 KmerIDs = Dict[Sequence, int]
 
+
+class ClusteringResult(object):
+    """
+    Stores the result of clustering as a ClusteredIDs object.
+    If clustering was not meaningful (e.g. one sequence per cluster),
+    returns a set of sequences instead that can be directly used as set of alternative
+    alleles of the variant site under construction.
+    """
+
+    def __init__(
+        self,
+        clustered_ids: Union[ClusteredIDs, None],
+        sequences: Union[Sequences, None],
+    ):
+        mutually_exclusive = (clustered_ids is None) ^ (sequences is None)
```

Comment:
This is very clever!!  
However, I had to look up this operator to remind myself which one it is (XOR), and then I had to look up XOR to remind myself of its outcome.  
```suggestion
        mutually_exclusive = (clustered_ids is None) is not (sequences is None)
```
might I suggest this just for clarity's sake?  
Again, what you had was very nice.

Reply:
Thx. I know what you mean, however I think we should keep this, because i)`is not` is not clearer to me, because what we want to enforce is that they can't both be None, nor both not be None. XOR is the exact definition of that. `is not` is harder to parse mentally ii)The surrounding code documents what's going on, the variable is called `mutually_exclusive` and the raised error if the XOR doesn't hold explains what's being enforced